src/components/interactive-builder/modals/question/question.modal.tsx

```diff
@@ -123,7 +123,17 @@ const QuestionModalContent: React.FC<QuestionModalProps> = ({
                       open={index === formField.questions.length - 1}
                       className={styles.obsGroupQuestionContent}
                     >
-                      <FormFieldProvider initialFormField={question} isObsGrouped={true}>
+                      <FormFieldProvider
+                        initialFormField={question}
+                        isObsGrouped={true}
+                        updateParentFormField={(updatedFormField) => {
+                          setFormField((prevFormField) => {
+                            const updatedQuestions = [...prevFormField.questions];
+                            updatedQuestions[index] = updatedFormField;
+                            return { ...prevFormField, questions: updatedQuestions };
+                          });
+                        }}
```

Comment:
It's better to define this as a separate function and pass it in, rather than defining it inside the component. Doing so prevents the function from being recreated on every render.
```
const updateParentFormField = (updatedFormField) => {
  setFormField((prevFormField) => {
  const updatedQuestions = [...prevFormField.questions];
  updatedQuestions[index] = updatedFormField;
  return { ...prevFormField, questions: updatedQuestions };
 });
}
```
```suggestion
                        updateParentFormField={updateParentFormField}
```